src/main/java/uk/gov/companieshouse/overseasentitiesapi/utils/PublicPrivateDataCombiner.java

```diff
@@ -108,11 +120,15 @@ public Map<String, Pair<PscApi, PrivateBoDataApi>> buildMergedBeneficialOwnerDat
    */
   public Map<String, Pair<CompanyOfficerApi, ManagingOfficerDataApi>> buildMergedManagingOfficerDataMap()
       throws ServiceException {
-    var combineMO = new HashMap<String, Pair<CompanyOfficerApi, ManagingOfficerDataApi>>();
+    this.combineMO = new HashMap<>();
 
     var privateMOs = privateDataRetrievalService.getManagingOfficerData();
     var publicOfficers = publicDataRetrievalService.getOfficers();
 
+    if (privateMOs == null || privateMOs.getManagingOfficerData() == null) {
```

Comment:
always fetch data

Reply:
I'd remove this if () return
and handle the left and/or right being null in the log method

Reply:
I'm unsure what this comment means 

This if statement was put in to prevent NullPointerExceptions further on in the code

Reply:
So the test testFilingGenerationWhenSuccessfulWithoutTrustsAndWithIdentityChecksForUpdate causes this to have a null pointer error if there are no null checks 

If you like I can wrap the for loops in the if statement instead of having it return a null

Reply:
So the test FilingServiceTest.testFilingGenerationWhenSuccessfulWithoutTrustsAndWithIdentityChecksForUpdate causes this to have a NullPointer Exception on the for loop of buildMergedManagingOfficerDataMap

Because of that, there does need to be some kind of if statement to check for null

I can wrap the for loop in an if statement if you think that would be better?

Reply:
Please mock getManagingOfficersData() to return a non-null value. We should not be fixing code to pass tests. We would not expect to return nulls when not in test?